src/plugin/vault/vault_test.go

```diff
@@ -225,51 +226,188 @@ func TestLogin(t *testing.T) {
+		t.Fatalf(err.Error())
+	}
+	if secret.LeaseDuration != 2 {
+		t.Fatalf("Expected pachyderm token with TTL=2s, but was %ds", secret.LeaseDuration)
+	}
+	pachToken := secret.Data["user_token"].(string)
+	reportedPachdAddress := secret.Data["pachd_address"].(string)
+	c, err := client.NewFromAddress(reportedPachdAddress)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	c.SetAuthToken(pachToken)
+	// Make sure token is valid
+	_, err = c.AuthAPIClient.GetAdmins(c.Ctx(), &auth.GetAdminsRequest{})
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	// Wait for TTL to expire and check that token is no longer valid
+	time.Sleep(time.Duration(secret.LeaseDuration+1) * time.Second)
 	_, err = c.AuthAPIClient.GetAdmins(c.Ctx(), &auth.GetAdminsRequest{})
 	if err == nil {
 		t.Fatalf("API call should fail, but token did not expire")
 	}
 }
 
+// If TTL is not in the request OR the config, make sure a sensible default is
+// used
+func TestLoginDefaultTTL(t *testing.T) {
+	c, _, secret := loginHelper(t, "")
+	if secret.LeaseDuration < 600 {
+		t.Fatalf("Expected Pachyderm token with duration at least 10m, but actual "+
+			"duration was %ds", secret.LeaseDuration)
+	}
+
+	// Make sure token is valid
+	_, err := c.AuthAPIClient.GetAdmins(c.Ctx(), &auth.GetAdminsRequest{})
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+}
+
+// TestLoginLongTTL tests that it's possible to get a Pachyderm token with a
+// long TTL
+func TestLoginLongTTL(t *testing.T) {
+	// Get Vault client
+	vaultClientConfig := vault.DefaultConfig()
+	vaultClientConfig.Address = vaultAddress
+	v, err := vault.NewClient(vaultClientConfig)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	v.SetToken("root")
+	vl := v.Logical()
+
+	// Get Pachyderm token with long TTL
+	secret, err := vl.Write(
+		fmt.Sprintf("/%v/login/github:bogusgithubusername", pluginName),
+		map[string]interface{}{"ttl": "768h"},
+	)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	if secret.LeaseDuration < int((700 * time.Hour).Seconds()) {
+		t.Fatalf("Expected pachyderm token with TTL=768h, but was %ds", secret.LeaseDuration)
+	}
+
+	// Make sure token is valid
+	pachToken, ok := secret.Data["user_token"].(string)
+	if !ok {
+		t.Fatalf("vault login response did not contain user token")
+	}
+	reportedPachdAddress, ok := secret.Data["pachd_address"].(string)
+	if !ok {
+		t.Fatalf("vault login response did not contain pachd address")
+	}
+	c, err := client.NewFromAddress(reportedPachdAddress)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	c.SetAuthToken(pachToken)
+	if _, err = c.AuthAPIClient.GetAdmins(c.Ctx(), &auth.GetAdminsRequest{}); err != nil {
+		t.Fatalf(err.Error())
+	}
+}
+
 func TestRenewBeforeTTLExpires(t *testing.T) {
 	ttl := 10
-	c, vaultClient, secret := loginHelper(t, fmt.Sprintf("%vs", ttl))
+	c, v, secret := loginHelper(t, fmt.Sprintf("%vs", ttl))
+	if secret.LeaseDuration < 2 {
+		t.Fatalf("expected lease to be at least 2s, but was: %d", secret.LeaseDuration)
+	} else if secret.LeaseDuration > 100 {
```

Comment:
I think we want to err if its greater than `10`, right?

Reply:
Fixed. I just wanted to make sure it wasn't 30 days, but 10s is indeed the value we expect